cmd/dep/glide_importer_test.go

```diff
@@ -34,6 +34,205 @@ func newTestContext(h *test.Helper) *dep.Ctx {
+		},
+		"exclude dir ignores mismatched package name": {
+			yaml: glideYaml{
+				Name:        "github.com/golang/mismatched-package-name",
+				ExcludeDirs: []string{"samples"},
+			},
+			projectRoot:         testGlideProjectRoot,
+			wantIgnoreCount:     1,
+			wantIgnoredPackages: []string{"github.com/golang/notexist/samples"},
+		},
+		"bad input, empty package name": {
+			yaml: glideYaml{
+				Imports: []glidePackage{{Name: ""}},
+			},
+			projectRoot:    testGlideProjectRoot,
+			wantConvertErr: true,
+		},
+	}
+
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+
+	ctx := newTestContext(h)
+	sm, err := ctx.SourceManager()
+	h.Must(err)
+	defer sm.Release()
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			g := newGlideImporter(discardLogger, true, sm)
+			g.yaml = testCase.yaml
+
+			if testCase.lock != nil {
+				g.lock = testCase.lock
+			}
+
+			manifest, lock, err := g.convert(testCase.projectRoot)
+			if err != nil {
+				if testCase.wantConvertErr {
+					return
+				}
+
+				t.Fatal(err)
+			}
+
+			// Lock checks.
+			if lock != nil && len(lock.P) != testCase.wantLockCount {
+				t.Fatalf("Expected lock to have %d project(s), got %d",
+					testCase.wantLockCount,
+					len(lock.P))
+			}
+
+			// Ignored projects checks.
+			if len(manifest.Ignored) != testCase.wantIgnoreCount {
+				t.Fatalf("Expected manifest to have %d ignored project(s), got %d",
+					testCase.wantIgnoreCount,
+					len(manifest.Ignored))
+			}
+
+			if !equalSlice(manifest.Ignored, testCase.wantIgnoredPackages) {
+				t.Fatalf("Expected manifest to have ignore %s, got %s",
+					strings.Join(testCase.wantIgnoredPackages, ", "),
+					strings.Join(manifest.Ignored, ", "))
+			}
+
+			// Constraints checks below. Skip if there is no want constraint.
+			if testCase.wantConstraint == "" {
+				return
+			}
+
+			d, ok := manifest.Constraints[testCase.projectRoot]
+			if !ok {
+				t.Fatalf("Expected the manifest to have a dependency for '%s' but got none",
+					testCase.projectRoot)
+			}
+
+			v := d.Constraint.String()
+			if v != testCase.wantConstraint {
+				t.Fatalf("Expected manifest constraint to be %s, got %s", testCase.wantConstraint, v)
+			}
+
+			p := lock.P[0]
+
+			if p.Ident().ProjectRoot != testCase.projectRoot {
+				t.Fatalf("Expected the lock to have a project for '%s' but got '%s'",
+					testCase.projectRoot,
+					p.Ident().ProjectRoot)
+			}
+
+			if p.Ident().Source != testCase.sourceRepo {
+				t.Fatalf("Expected locked source to be %s, got '%s'", testCase.sourceRepo, p.Ident().Source)
+			}
+
+			lv := p.Version()
+			lpv, ok := lv.(gps.PairedVersion)
+
+			if !ok {
+				if testCase.matchPairedVersion {
+					t.Fatalf("Expected locked version to be PairedVersion but got %T", lv)
+				}
+
+				return
+			}
+
+			ver := lpv.String()
+			if ver != testCase.wantVersion {
+				t.Fatalf("Expected locked version to be '%s', got %s", testCase.wantVersion, ver)
+			}
+
+			if testCase.wantRevision != nil {
```

Comment:
Since we only check `wantRevision` when `matchPairedVersion` is set, how about simplifying it to a `Revision` struct, not pointer. Then `revisionPtr` can be removed as well.